### PR TITLE
Test and fix cook notebook, some perf improvements

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/Dofs/DirichletBoundaryConditions.jl [hunk 1]
```diff
@@ -289,32 +301,6 @@ function zero_out_columns!(K, dofs::Vector{Int}) # can be removed in 0.7 with #2
     end
 end
 
-function zero_out_rows!(K, dofs::Vector{Int})
```
Comment (KristofferC) [inline src/Dofs/DirichletBoundaryConditions.jl]: Base was just as fast as me for this

File: src/Dofs/DofHandler.jl [hunk 2]
```diff
@@ -240,23 +240,37 @@ function _create_sparsity_pattern(dh::DofHandler, sym::Bool)
...
+    cnt = 0
     for element_id in 1:ncells
         celldofs!(global_dofs, dh, element_id)
         @inbounds for j in 1:n, i in 1:n
             dofi = global_dofs[i]
             dofj = global_dofs[j]
             sym && (dofi > dofj && continue)
-            push!(I, dofi)
-            push!(J, dofj)
+            cnt += 1
```
Comment (KristofferC) [inline src/Dofs/DofHandler.jl]: avoids calling into `push!`:

Before:

```
  1.531062 seconds (24 allocations: 848.220 MiB, 24.73% gc time)
  1.568128 seconds (24 allocations: 848.220 MiB, 27.00% gc time)
  1.903661 seconds (24 allocations: 848.220 MiB, 25.30% gc time)
  1.637734 seconds (24 allocations: 848.220 MiB, 29.55% gc time)
```

After

```
  1.267178 seconds (24 allocations: 848.220 MiB, 33.20% gc time)
  1.243249 seconds (24 allocations: 848.220 MiB, 34.37% gc time)
  1.200653 seconds (24 allocations: 848.220 MiB, 33.75% gc time)
  1.215593 seconds (24 allocations: 848.220 MiB, 33.97% gc time)
```

File: src/Dofs/DirichletBoundaryConditions.jl [hunk 3]
```diff
@@ -267,7 +270,16 @@ function apply!(KK::Union{SparseMatrixCSC,Symmetric}, f::AbstractVector, dbcs::D
         end
     end
     zero_out_columns!(K, dbcs.prescribed_dofs)
-    zero_out_rows!(K, dbcs.prescribed_dofs)
+    if strategy == APPLY_TRANSPOSE
+        KK = copy(K)
```
Comment (fredrikekre) [inline src/Dofs/DirichletBoundaryConditions.jl]: Byt namn.

Comment (KristofferC) [reply src/Dofs/DirichletBoundaryConditions.jl]: Done!!!!!!!

**Pull request merged**